src/v/rpc/simple_protocol.cc

```diff
@@ -222,8 +222,9 @@ simple_protocol::dispatch_method_once(header h, net::server::resources rs) {
                         reply_buf.set_version(ctx->get_header().version);
                     }
                     return send_reply(ctx, std::move(reply_buf))
-                      .finally([m, l = std::move(l)]() mutable {
-                          m->probes.latency_hist().record(std::move(l));
+                      .finally([ctx, m, l = std::move(l)]() mutable {
+                          ctx->res.hist().record(l);
```

Comment:
I still don't understand: `l = ctx->res.hist().auto_measure()`.

`auto_measure()` is:
```c+++
std::unique_ptr<hdr_hist::measurement> hdr_hist::auto_measure() {
    return std::make_unique<measurement>(*this);
}
```
`measurement` destructor is:
```c++
        ~measurement() noexcept {
            if (!_detached) {
                _h.get()._probes.erase(_h.get()._probes.iterator_to(*this));
                // !detached && trace
                // do not move outside of this nested if
                if (_trace) {
                    _h.get().record(compute_duration_micros());
                }
            }
        }
```
So why is this line needed? Doesn't it just result in double counting?

Reply:
Ohh I see you're point now. Let me test.